vec120/src/main/java/com/foursoft/vecmodel/vec120/navigations/DescriptionNavs.java

```diff
@@ -0,0 +1,117 @@
+/*-
+ * ========================LICENSE_START=================================
+ * vec120
+ * %%
+ * Copyright (C) 2020 - 2022 4Soft GmbH
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.vecmodel.vec120.navigations;
+
+import com.foursoft.vecmodel.common.HasDescription;
+import com.foursoft.vecmodel.vec120.VecAbstractLocalizedString;
+import com.foursoft.vecmodel.vec120.VecLanguageCode;
+import com.foursoft.vecmodel.vec120.VecLocalizedTypedString;
+import com.foursoft.vecmodel.vec120.predicates.VecPredicates;
+import com.foursoft.vecmodel.vec120.utils.StreamUtils;
+import com.foursoft.vecmodel.vec120.utils.StringUtils;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Navigation methods for {@link HasDescription} with {@link VecAbstractLocalizedString}s.
+ */
+public final class DescriptionNavs {
+
+    private DescriptionNavs() {
+        // hide default constructor
+    }
+
+    public static Function<HasDescription<? extends VecAbstractLocalizedString>, Optional<String>> germanLanguageStringByDescription() {
+        return languageStringByDescription(VecLanguageCode.DE);
+    }
+
+    public static Function<HasDescription<? extends VecAbstractLocalizedString>, Optional<String>> englishLanguageStringByDescription() {
+        return languageStringByDescription(VecLanguageCode.EN);
+    }
+
+    public static Function<HasDescription<? extends VecAbstractLocalizedString>, Optional<String>> languageStringByDescription(
```

Comment:
what do you think of the methode name "descriptionBy"

Reply:
I think `descriptionIn/With(VecLanguageCode languageCode)` makes more sense in this context. What do you think?

Reply:
yes you are right. i would say "in"

Reply:
Another question would be if the typed methods could get a better name as well.

For example shortening `germanLanguageString` to `germanString` and `germanTypedStringBy`. What do you think of that?